src/canvas/renderCanvas.js

```diff
@@ -99,206 +96,290 @@ function renderReactMarksAndMeasures(
-          fillStyle = COLORS.REACT_LAYOUT_EFFECTS;
-          hoveredFillStyle = COLORS.REACT_LAYOUT_EFFECTS_HOVER;
-          groupSelectedFillStyle = COLORS.REACT_LAYOUT_EFFECTS_SELECTED;
-          break;
-        case 'passive-effects':
-          fillStyle = COLORS.REACT_PASSIVE_EFFECTS;
-          hoveredFillStyle = COLORS.REACT_PASSIVE_EFFECTS_HOVER;
-          groupSelectedFillStyle = COLORS.REACT_PASSIVE_EFFECTS_SELECTED;
-          break;
-        default:
-          console.warn(`Unexpected type "${type}"`);
-          break;
-      }
-
-      y = baseY + REACT_WORK_SIZE * depth + REACT_GUTTER_SIZE * depth - offsetY;
-
-      // $FlowFixMe We know these won't be null
-      context.fillStyle = showHoverHighlight
-        ? hoveredFillStyle
-        : showGroupHighlight
-        ? groupSelectedFillStyle
-        : fillStyle;
-      context.fillRect(
-        Math.floor(x),
-        Math.floor(y),
-        Math.floor(width),
-        REACT_WORK_SIZE,
-      );
+      fillStyle = COLORS.REACT_LAYOUT_EFFECTS;
+      hoveredFillStyle = COLORS.REACT_LAYOUT_EFFECTS_HOVER;
+      groupSelectedFillStyle = COLORS.REACT_LAYOUT_EFFECTS_SELECTED;
       break;
-    case 'schedule-render':
-    case 'schedule-state-update':
-    case 'suspend':
-      const {isCascading} = ((eventOrMeasure: any): ReactEvent);
-
-      x = timestampToPosition(timestamp, state);
-      if (x + EVENT_SIZE / 2 < 0 || canvasWidth < x) {
-        return; // Not in view
-      }
-
-      switch (type) {
-        case 'schedule-render':
-        case 'schedule-state-update':
-          if (isCascading) {
-            fillStyle = showHoverHighlight
-              ? COLORS.REACT_SCHEDULE_CASCADING_HOVER
-              : COLORS.REACT_SCHEDULE_CASCADING;
-          } else {
-            fillStyle = showHoverHighlight
-              ? COLORS.REACT_SCHEDULE_HOVER
-              : COLORS.REACT_SCHEDULE;
-          }
-          break;
-        case 'suspend':
-          fillStyle = showHoverHighlight
-            ? COLORS.REACT_SUSPEND_HOVER
-            : COLORS.REACT_SUSPEND;
-          break;
-        default:
-          console.warn(`Unexpected event or measure type "${type}"`);
-          break;
-      }
-
-      if (fillStyle !== null) {
-        const circumference = REACT_EVENT_SIZE;
-        y = baseY + REACT_EVENT_SIZE / 2 - offsetY;
-
-        context.beginPath();
-        context.fillStyle = fillStyle;
-        context.arc(x, y, circumference / 2, 0, 2 * Math.PI);
-        context.fill();
-      }
+    case 'passive-effects':
+      fillStyle = COLORS.REACT_PASSIVE_EFFECTS;
+      hoveredFillStyle = COLORS.REACT_PASSIVE_EFFECTS_HOVER;
+      groupSelectedFillStyle = COLORS.REACT_PASSIVE_EFFECTS_SELECTED;
       break;
     default:
-      console.warn(`Unexpected event or measure type "${type}"`);
+      console.warn(`Unexpected type "${type}"`);
       break;
   }
+
+  const y = baseY - offsetY;
+
+  // $FlowFixMe We know these won't be null
```

Comment:
This `$FlowFixMe` doesn't actually reach to the  ternary clauses because it only matches literally next code line.

I'd personally solve by throwing instead of console.warning on line 227, allowing flow to refine the types to always strings.